.github/workflows/testing.yml

```diff
@@ -26,12 +26,12 @@ jobs:
 
       - name: Install package
         run: |
-          pip install -e .
+          pip install .
```

Comment:
`-e` has been deprecated

Reply:
When was `-e` of pip deprecated?

Reply:
Oh I see. `pip install -e` is not deprecated with `pyproject.toml`, so the changes around removing `-e` could be resolved by using `pyproject.toml`.

Reply:
Editable mode is a setuptools feature that might not be availabe in other installation backends (e.g. poetry).  pip just relies to the instructions to the backend:

https://pip.pypa.io/en/stable/topics/local-project-installs/#editable-installs

But I think most backends implement this. Now, the real problematic part is that editable mode in setuptools changed dramatically on recent versions:

https://setuptools.pypa.io/en/latest/userguide/development_mode.html

My suggestion was that we should not use that API in the CI because it is unstable (within setuptools versions and across build backends implementations) and I suspected it was generating an error (I don't remember if I was right). This is in line what what setuptools suggest themselves (see the attention section in top of limitations here):
https://setuptools.pypa.io/en/latest/userguide/development_mode.html#legacy-behavior

Reply:
Interesting. Thanks for the additional information @h-mayorquin ! In the pynwb/hdmf stack, we use `pip install -e 
 . ` to help `coverage` find the source files, but we can do it differently as @CodyCBakerPhD has done here. We'll make the same changes on the pynwb/hdmf stack.